sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -247,6 +252,38 @@ class Analyzer(
     }
   }
 
+  object ResolvePivot extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case p: Pivot if !p.childrenResolved => p
+      case Pivot(groupByExprs, pivotColumn, pivotValues, aggregates, child) =>
+        val singleAgg = aggregates.size == 1
+        val pivotAggregates: Seq[NamedExpression] = pivotValues.flatMap{ value =>
+          aggregates.map{ aggregate =>
+            val filteredAggregate = aggregate.transformDown{
+              case u: UnaryExpression if u.isInstanceOf[AggregateExpression] =>
+                u.withNewChildren(Seq(
+                  If(EqualTo(pivotColumn, Literal(value)), u.child, Literal(null))
+                ))
```

Comment:
@aray We have switched to our new `AggregateExpression` and `AggregateFunction` interfaces. With the new interface, we always use `AggregateExpression` to wrap an `AggregateFunction`. So, I think the matching condition can be changed to match `AggregateFunction` instead of matching `AggregateExoression`.